splat.pl

```diff
@@ -0,0 +1,953 @@
+  
+Examples:
+
+Note the use of splat instead of splat.pl.  This is required to grab the
+necessary ACA EGSE perl libraries.
+
+From solaris in ASCDS environment (to get correct GMT):
+
+  /proj/telmon/aca/telem_tools/dump2occ.pl telem.sto.gz > telem.occ
+
+From linux:
+
+  splat -i telem.occ
+
+  gunzip --stdout telem.sto.gz \
+  | $SKA/data/aca_egse/telem_convert/ehs5tdsn.pl \
+  | splat
+
+USAGE
+  ; 
+
+  $ACA_DECOM_HOME = "$ENV{SKA}/decom/aca";
+  $MAX_EVENT_LOG_LINES = 10000;
+  $opt_page = 1;
+  $opt_time = 0;
+  $opt_vcdu = 0;
+  $opt_raw  = 1;
+  $opt_i    = undef;
+  $opt_image= 1;
+  $opt_h    = 0;
+  $opt_tctm = 0;
+  $opt_logsc= 1;
+  $opt_pea  = 'a';
+  $opt_pause= 0;
+  $opt_scale  = 1;
+  
+  &GetOptions ('h',		# Help
+	       'page!',		# Run in display page mode
+	       'image!',	# Display images
+	       'raw!',		# Display raw data
+	       'i=s',		# Optional input file
+	       'time=i',	# Min time between updates
+	       'vcdu=i',	# Minimum VCDU counter value
+	       'tctm!',		# TCTM format?
+	       'logsc!',	# Display log-scale grey-scale images
+	       'out=s',		# Write ACA data from slots to an ASCII file
+	       'pea=s',		# PEA ("a" or "b")
+               'pause!',        # Start out paused
+               'scale!',        # Apply pixel signal scaling: DN = (scale_factor/32 * pixsig - 50)
+	      );
+  
+  die $Usage if ($opt_h);
+  
+  $opt_time = $opt_time / 1000.0; # convert to seconds
+}
+
+##***************************************************************************
+sub set_info_table {
+##***************************************************************************
+    my $slt;
+    my $j;
+    my $txt;
+    for $slt (1 .. 8) {
+	for $j (0 .. $#info_msid) {
+	    if (defined $cal{$info_msid[$j]}[$slt-1]) {
+		$txt = sprintf($info_format[$j], $cal{$info_msid[$j]}[$slt-1]);
+		if ($txt ne $info_table_txt[$j][$slt]) {
+		  $info_table_txt[$j][$slt] = $txt;
+		}
+	    }
+	}
+    }
+}
+
+##***************************************************************************
+sub process_frame {
+##***************************************************************************
+#   Get vcdu at *beginning* of 4 ACA minor frames 
+  my $fast = shift;
+
+  my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = gmtime($tlm_time);
+  $tlm_gmt = sprintf "%02d:%02d:%02d",$hour,$min,$sec;
+  $tlm_date = sprintf "%4d:%03d", $year+1900,$yday+1;
+ 
+  return if $fast;
+   
+    # check if min VCDU reached
+
+#    if ($vcdu < $opt_vcdu) {
+#	print $CURSOR_HOME, "VCDU = $vcdu\n";
+#	return;
+#    }
+
+    if ($vcdu % 4 == 0) {	# should have a complete aca packet (224 bytes)
+	if ($opt_time > 0) {
+	    $dt = time - $time0;
+	    sleep ($opt_time - $dt) if ($dt < $opt_time);
+	    $time0 = time;
+	}
+	print $CURSOR_HOME;
+        printf "Format:%2d  VCDU:%9d (%6d,%3.3d)  ",$format,$vcdu,$vcdu/128,$vcdu%128; 
+        if ($opt_tctm) {
+	    my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday) = gmtime($utc1970);
```

Comment:
Is this utc1970 supposed to do anything?  I don't know what tctm is in this context, but unix times would probably work with the tctm flag if there were a line like 
```
$utc1970 = time;
```
around here.